### DISCUSSION TITLE
Invalidate a query cache by running it with the `ttl` set to `0`?

### DISCUSSION THREAD

**Discussion by tavoyne** [prisma/prisma#25200] [Q&A]
### Question

Say I have this query that's run frequently:

```tsx
const users = await db.user.findMany({
  cacheStrategy: { ttl: 300 },
});
```

Does the same query with a `ttl` of `0`:

```tsx
const users = await db.user.findMany({
  cacheStrategy: { ttl: 0 },
});
```

...invalidate the cache value for the first query?

**Comment by ludralph** [comment]
Hi @tavoyne 👋 

Setting `ttl` to 0 does not invalidate the cache.  The `ttl` is the duration in seconds for which a cached response is considered fresh.
The code above means that the query will bypass the cache and fetch fresh data from the database, but it will not invalidate or remove the existing cache entries created by the previous query.

For Prisma Accelerate, you can currently [invalidate the cache](https://www.prisma.io/docs/accelerate/faq#how-can-i-invalidate-a-cache-on-accelerate) at a project level.

**Reply by tavoyne** [reply]
> [...] but it will not invalidate or remove the existing cache entries created by the previous query.

Why not, since it's the same query hence the same cache key?

I'm currently testing it and it seems to be working fine.

**Reply by ludralph** [reply]
Hi @tavoyne 

I tested this and you're correct that it does.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
